internal/controller/dnsrecord_controller.go

```diff
@@ -89,56 +97,61 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	}
 
 	if !controllerutil.ContainsFinalizer(dnsRecord, DNSRecordFinalizer) {
+		dnsRecord.Status.QueuedFor = metav1.NewTime(reconcileStart.Add(validationRequeueTime))
 		logger.Info("Adding Finalizer", "name", DNSRecordFinalizer)
 		controllerutil.AddFinalizer(dnsRecord, DNSRecordFinalizer)
 		err = r.Update(ctx, dnsRecord)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
-		return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{RequeueAfter: validationRequeueTime}, nil
 	}
 
 	var reason, message string
-	status := metav1.ConditionTrue
-	reason = "ProviderSuccess"
-	message = "Provider ensured the dns record"
 	err = dnsRecord.Validate()
 	if err != nil {
-		status = metav1.ConditionFalse
 		reason = "ValidationError"
 		message = fmt.Sprintf("validation of DNSRecord failed: %v", err)
-		setDNSRecordCondition(dnsRecord, string(conditions.ConditionTypeReady), status, reason, message)
-		return r.updateStatus(ctx, previous, dnsRecord)
+		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse, reason, message)
+		return r.updateStatus(ctx, previous, dnsRecord, "0s")
```

Comment:
not a big fan of this being a string an being parsed later on. If we have to do it this way, perhaps we could also pull it out to a constant? `requeueImmediate`  or something?

Reply:
it seemed like that as well for me. [Here](https://github.com/Kuadrant/dns-operator/pull/72/commits/4f4f387d7c369fb04baca3c2ce8e097a88c3fc4d#diff-adacb6a4589c65c2d853c0f59ca02d6b8fb50bb388b41de1d13e99b0d128cfe1R51) is the way it looks good to me. Could have those changes sin this PR if you prefer it better. But all of them should land soon so i decided to keep this one as is :thinking:

Reply:
and along with it there would be only one `parse` call and it is only being set from `time.Duration.String()` string so we could be sure it will succeed

Reply:
Ok in that case I am ok to approve with the improvements that are coming

Reply:
well, there is no harm in having them here I guess.